src/main/java/org/xwiki/git/GitManager.java

```diff
@@ -68,6 +69,23 @@ default Repository getRepository(String repositoryURI, String localDirectoryName
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * Clone a Git repository as Bare using the CloneCommand provided by user and store it locally in the
+     * XWiki Permanent directory. If the repository is already cloned, no action is done.
+     *
+     * @param repositoryURI the URI to the Git repository to clone (eg "git://github.com/xwiki/xwiki-commons.git")
+     * @param localDirectoryName the name of the directory where the Git repository will be cloned (this directory is
+     *        relative to the permanent directory
+     * @param cloneCommand the clone settings used as CloneCommand object
+     * @return the cloned Repository instance
+     * @since 9.10
+     */
+    @Unstable
+    default Repository getRepositoryBare(String repositoryURI, String localDirectoryName, CloneCommand cloneCommand)
```

Comment:
I would not name it `getRepositoryBare`  but `getRepository`  so that it's generic. It's also not right to both pass a CloneCommand and set it in the code. The passed CloneCommand is supposed to be readonly (final).

Could you please change that? f you really think it's good to have a `getRepositoryBare` API, then the signature should be `getRepositoryBare(String repositoryURI, String localDirectoryName)` + `getRepositoryBare(String repositoryURI, String localDirectoryName, String username, String accessCode)`.

Reply:
Yes `getRepository` is better here. We also won't need the private method (since it will be public method which can be used by other overloads in return).